sklearn/metrics/_pairwise_distances_reduction/_argkmin.pyx.tp

```diff
@@ -66,13 +67,14 @@ cdef class ArgKmin{{name_suffix}}(BaseDistancesReduction{{name_suffix}}):
         """
         if (
             metric in ("euclidean", "sqeuclidean")
-            and not issparse(X)
-            and not issparse(Y)
+            and not (issparse(X) ^ issparse(Y))
         ):
             # Specialized implementation with improved arithmetic intensity
             # and vector instructions (SIMD) by processing several vectors
-            # at time to leverage a call to the BLAS GEMM routine as explained
-            # in more details in the docstring.
+            # at time to leverage a call to a matrix multiplication routine.
+            # Decomposing the euclidean distance into as a sum of squared
+            # matrices and a middle term cross product is explained in more
+            # details in the docstring of MiddleTermComputer.
```

Comment:
I can't suggest, but I think we can be implementation details agnostic here and move the part about SIMD in `DenseDenseMiddleTermComputer`'s docstring.
```
            # Specialized implementation of ArgKmin for the Euclidean distance.
            # This implementation processes the computation of the chunks
            # of the distance using a decomposition of the chunks of the
            # distance matrix for the Squared Euclidean distance.
            # This specialisation has an improved arithmetic intensity for
            # both the dense and the sparse settings, allowing in most cases
            # speed-ups of several orders compare the generic Argkmin implementation.
            # For more information, see MiddleTermComputer.
```

Reply:
Thanks, how to write good comments is not totally obvious to me